From: Fabrice Olivero <f-olivero@ti.com>
Date: Fri, 8 Apr 2011 14:52:49 +0000 (+0200)
Subject: OMAP4 DSS HDMI: Disable data islands packets in DVI mode
X-Git-Tag: android-2.6.35-2.3-omap4.13.1~41
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=14cb8265af1d85a1429880a7e833318169ea9b36

OMAP4 DSS HDMI: Disable data islands packets in DVI mode

This is following IP specification recommendation but should not change
behavior of DVI mode.
Data islands packets need to be sent only in DVI mode.

Change-Id: I3ff574d588e264443047d50e4f058be78d182d57
Signed-off-by: Fabrice Olivero <f-olivero@ti.com>
Signed-off-by: Srinvias Pulukuru <srinivas.pulukuru@ti.com>
---

diff --git a/arch/arm/plat-omap/hdmi_lib.c b/arch/arm/plat-omap/hdmi_lib.c
index f664421..8d10eb7 100644
--- a/arch/arm/plat-omap/hdmi_lib.c
+++ b/arch/arm/plat-omap/hdmi_lib.c
@@ -1664,18 +1664,38 @@ int hdmi_lib_enable(struct hdmi_config *cfg)
 		repeat_param.GenericPacketRepeat = PACKETREPEATON;
 	}
 
-	/* enable/repeat the infoframe */
-	repeat_param.AVIInfoFrameED = PACKETENABLE;
-	repeat_param.AVIInfoFrameRepeat = PACKETREPEATON;
-	/* wakeup */
-	repeat_param.AudioPacketED = PACKETENABLE;
-	repeat_param.AudioPacketRepeat = PACKETREPEATON;
-	/* ISCR1 transmission */
-	repeat_param.MPEGInfoFrameED = PACKETDISABLE;
-	repeat_param.MPEGInfoFrameRepeat = PACKETREPEATOFF;
-	/* ACP transmission */
-	repeat_param.SPDInfoFrameED = cfg->supports_ai;
-	repeat_param.SPDInfoFrameRepeat = cfg->supports_ai;
+
+	if (cfg->hdmi_dvi == 1) {
+		/* HDMI mode */
+
+		/* enable/repeat the infoframe */
+		repeat_param.AVIInfoFrameED = PACKETENABLE;
+		repeat_param.AVIInfoFrameRepeat = PACKETREPEATON;
+		/* wakeup */
+		repeat_param.AudioPacketED = PACKETENABLE;
+		repeat_param.AudioPacketRepeat = PACKETREPEATON;
+		/* ISCR1 transmission */
+		repeat_param.MPEGInfoFrameED = PACKETDISABLE;
+		repeat_param.MPEGInfoFrameRepeat = PACKETREPEATOFF;
+		/* ACP transmission */
+		repeat_param.SPDInfoFrameED = cfg->supports_ai;
+		repeat_param.SPDInfoFrameRepeat = cfg->supports_ai;
+	} else {
+		/* DVI mode */
+
+		/* enable/repeat the infoframe */
+		repeat_param.AVIInfoFrameED = PACKETDISABLE;
+		repeat_param.AVIInfoFrameRepeat = PACKETREPEATOFF;
+		/* wakeup */
+		repeat_param.AudioPacketED = PACKETDISABLE;
+		repeat_param.AudioPacketRepeat = PACKETREPEATOFF;
+		/* ISCR1 transmission */
+		repeat_param.MPEGInfoFrameED = PACKETDISABLE;
+		repeat_param.MPEGInfoFrameRepeat = PACKETREPEATOFF;
+		/* ACP transmission */
+		repeat_param.SPDInfoFrameED = PACKETDISABLE;
+		repeat_param.SPDInfoFrameRepeat = PACKETREPEATOFF;
+	}
 
 	r = hdmi_core_av_packet_config(av_name, repeat_param);
 